XMonad/Layout/CircleEx.hs

```diff
@@ -0,0 +1,161 @@
+{-# LANGUAGE FlexibleInstances, MultiParamTypeClasses #-}
+{-# LANGUAGE RecordWildCards #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Layout.CircleEx
+-- Description :  An elliptical, overlapping layout - extended version.
+-- Copyright   :  (c) Peter De Wachter, Ilya V. Portnov
+-- License     :  BSD-style (see LICENSE)
+--
+-- Maintainer  :  Ilya V. Portnov <portnov84@rambler.ru>
+-- Stability   :  unstable
+-- Portability :  unportable
+--
+-- Circle is an elliptical, overlapping layout. Original code by Peter De Wachter,
+-- extended by Ilya Porntov.
+-----------------------------------------------------------------------------
+
+module XMonad.Layout.CircleEx (
+    -- * Usage
+    -- $usage
+    CircleEx, circleEx,
+    cNMaster, cMasterRatio, cSlaveRatio, cMultiplier, cDelta,
+    CircleExMsg (..)
+  )
+  where
+
+import Data.Ratio
+
+import XMonad.Prelude
+import XMonad
+import qualified XMonad.StackSet as W
+
+-- $usage
+-- The layout puts first N windows (called master) to the center of screen. 
+-- Others (called slave) are organized in a circle (well, ellipse). Size of
```

Comment:
I'd `s/slave/stack` throughout the whole module. (I've been wanting to do this across contrib for a while, but haven't gotten around to it)

Reply:
Hm, I'm okay, but the word "stack" does not seem to describe "other-than-master" windows to me. As far as I understand, in xmonad stack are all windows except floating ones.
Maybe "secondary" instead?

Reply:
> As far as I understand, in xmonad stack are all windows except floating ones.

This is technically true, but at this point "stack windows" seems to be synonymous with "non-master windows" in contrib. I don't have any strong opiniont on this, though, "secondary" sounds fine as well